test/common/upstream/eds_speed_test.cc

```diff
@@ -0,0 +1,202 @@
+public:
+  EdsSpeedTest() : api_(Api::createApiForTest(stats_)) { resetCluster(); }
+
+  void resetCluster() {
+    resetCluster(R"EOF(
+      name: name
+      connect_timeout: 0.25s
+      type: EDS
+      lb_policy: ROUND_ROBIN
+      eds_cluster_config:
+        service_name: fare
+        eds_config:
+          api_config_source:
+            api_type: REST
+            cluster_names:
+            - eds
+            refresh_delay: 1s
+    )EOF",
+                 Cluster::InitializePhase::Secondary);
+  }
+
+  void resetClusterDrainOnHostRemoval() {
+    resetCluster(R"EOF(
+        name: name
+        connect_timeout: 0.25s
+        type: EDS
+        lb_policy: ROUND_ROBIN
+        drain_connections_on_host_removal: true
+        eds_cluster_config:
+          service_name: fare
+          eds_config:
+            api_config_source:
+              api_type: REST
+              cluster_names:
+              - eds
+              refresh_delay: 1s
+    )EOF",
+                 Cluster::InitializePhase::Secondary);
+  }
+
+  void resetClusterLoadedFromFile() {
+    resetCluster(R"EOF(
+      name: name
+      connect_timeout: 0.25s
+      type: EDS
+      lb_policy: ROUND_ROBIN
+      eds_cluster_config:
+        eds_config:
+          path: "eds path"
+    )EOF",
+                 Cluster::InitializePhase::Primary);
+  }
+
+  void resetCluster(const std::string& yaml_config, Cluster::InitializePhase initialize_phase) {
+    local_info_.node_.mutable_locality()->set_zone("us-east-1a");
+    eds_cluster_ = parseClusterFromV2Yaml(yaml_config);
+    Envoy::Stats::ScopePtr scope = stats_.createScope(fmt::format(
+        "cluster.{}.",
+        eds_cluster_.alt_stat_name().empty() ? eds_cluster_.name() : eds_cluster_.alt_stat_name()));
+    Envoy::Server::Configuration::TransportSocketFactoryContextImpl factory_context(
+        admin_, ssl_context_manager_, *scope, cm_, local_info_, dispatcher_, random_, stats_,
+        singleton_manager_, tls_, validation_visitor_, *api_);
+    cluster_.reset(
+        new EdsClusterImpl(eds_cluster_, runtime_, factory_context, std::move(scope), false));
+    EXPECT_EQ(initialize_phase, cluster_->initializePhase());
+    eds_callbacks_ = cm_.subscription_factory_.callbacks_;
+  }
+
+  void initialize() {
+    EXPECT_CALL(*cm_.subscription_factory_.subscription_, start(_));
+    cluster_->initialize([this] { initialized_ = true; });
+  }
+
+  void doOnConfigUpdateVerifyNoThrow(
+      const envoy::config::endpoint::v3::ClusterLoadAssignment& cluster_load_assignment) {
+    Protobuf::RepeatedPtrField<ProtobufWkt::Any> resources;
+    resources.Add()->PackFrom(cluster_load_assignment);
+    VERBOSE_EXPECT_NO_THROW(eds_callbacks_->onConfigUpdate(resources, ""));
+  }
+
+  // Set up an EDS config with multiple priorities, localities, weights and make sure
+  // they are loaded and reloaded as expected.
+  void PriorityAndLocalityWeightedHelper(bool ignore_unknown_dynamic_fields, int num_hosts) {
```

Comment:
Please use Envoy style for naming, i.e. `priorityAndLocality..`

Reply:
Done.